src/dynd/func/arithmetic.cpp

```diff
@@ -4,73 +4,73 @@
 //
 
 #include <dynd/func/arithmetic.hpp>
+#include <dynd/func/elwise.hpp>
 #include <dynd/kernels/arithmetic.hpp>
-#include <dynd/array.hpp>
 
 using namespace dynd;
 
-#define BUILTIN_ROW(NAME, A0)                                                  \
-  {                                                                            \
-    NULL, &create<nd::NAME<A0, int8_t>>, &create<nd::NAME<A0, int16_t>>,       \
-        &create<nd::NAME<A0, int32_t>>, &create<nd::NAME<A0, int64_t>>, NULL,  \
-        &create<nd::NAME<A0, uint8_t>>, &create<nd::NAME<A0, uint16_t>>,       \
-        &create<nd::NAME<A0, uint32_t>>, &create<nd::NAME<A0, uint64_t>>,      \
-        NULL, NULL, &create<nd::NAME<A0, float>>,                              \
-        &create<nd::NAME<A0, double>>, NULL,                                   \
-        &create<nd::NAME<A0, dynd::complex<float>>>,                           \
-        &create<nd::NAME<A0, dynd::complex<double>>>,                          \
-  }
-
-#define BUILTIN_TABLE(NAME)                                                    \
-  const create_t nd::NAME::builtin_table[builtin_type_id_count -               \
-                                         2][builtin_type_id_count - 2] = {     \
-      {NULL},                                                                  \
-      BUILTIN_ROW(NAME##_ck, int8_t),                                          \
-      BUILTIN_ROW(NAME##_ck, int16_t),                                         \
-      BUILTIN_ROW(NAME##_ck, int32_t),                                         \
-      BUILTIN_ROW(NAME##_ck, int64_t),                                         \
-      {NULL},                                                                  \
-      BUILTIN_ROW(NAME##_ck, uint8_t),                                         \
-      BUILTIN_ROW(NAME##_ck, uint16_t),                                        \
-      BUILTIN_ROW(NAME##_ck, uint32_t),                                        \
-      BUILTIN_ROW(NAME##_ck, uint64_t),                                        \
-      {NULL},                                                                  \
-      {NULL},                                                                  \
-      BUILTIN_ROW(NAME##_ck, float),                                           \
-      BUILTIN_ROW(NAME##_ck, double),                                          \
-      {NULL},                                                                  \
-      BUILTIN_ROW(NAME##_ck, dynd::complex<float>),                            \
-      BUILTIN_ROW(NAME##_ck, dynd::complex<double>)};
-
-BUILTIN_TABLE(add);
-BUILTIN_TABLE(sub);
-BUILTIN_TABLE(mul);
-BUILTIN_TABLE(div);
-
-#undef BUILTIN_TABLE
-#undef BUILTIN_ROW
+nd::arrfunc nd::plus::make()
+{
+  return functional::elwise(as_arrfunc<multidispatch_plus_ck>());
+}
+
+struct nd::plus nd::plus;
+
+nd::array nd::operator+(const nd::array &a0) { return nd::plus(a0); }
```

Comment:
So nd::plus is both the type of the class and the instance, is that right? I'm used to that kind of thing being problematic, does it work fully consistently now?